crates/router/src/connector/square.rs

```diff
@@ -844,7 +844,22 @@ impl api::IncomingWebhook for Square {
         _merchant_id: &str,
         _secret: &[u8],
     ) -> CustomResult<Vec<u8>, errors::ConnectorError> {
-        Ok(format!("{}{}", request.uri, String::from_utf8_lossy(request.body)).into_bytes())
+        let authority = request
+            .headers
+            .get(actix_web::http::header::HOST)
+            .map(|header_value| header_value.to_str().map(String::from))
+            .transpose()
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookResponseEncodingFailed)?
+            .unwrap_or("".to_string());
```

Comment:
```suggestion
       let header_value = request
            .headers
            .get(actix_web::http::header::HOST)
            .ok_or(errors::ConnectorError::WebhookSourceVerificationFailed)?;
        let authority = header_value
            .to_str()
            .into_report()
            .change_context(errors::ConnectorError::WebhookResponseEncodingFailed)?;
```

Let's not use unwrap_or, if not able to get the header let's stop the flow

Reply:
Done